docs/jwst/extract_2d/reference_files.rst

```diff
@@ -1,26 +1,91 @@
 Reference Files
 ===============
 
-.. include:: ../includes/standard_keywords.rst
+The `extract_2d` step uses WAVECORR and WAVELENGTHRANGE reference files.
+The WAVECORR reference file is only used for NIRSpec fixed-slit and MOS
+exposures.
+The WAVELENGTHRANGE reference file is only used for NIRCam and NIRISS
+Wide-Field Slitless Spectroscopy (WFSS) exposures.
+
+WAVECORR Reference File
+-----------------------
 
-WAVECORR Reference
-------------------
+:REFTYPE: WAVECORR
+:Data model: `WaveCorrModel`
 
-To apply the Nirspec wavelength zero-point correction, this step uses the
-WAVECORR reference file. The zero-point correction is applied to observations
-with EXP_TYPE of "NRS_FIXEDSLIT", "NRS_BRIGHTOBJ" or "NRS_MSASPEC". This is an
-optional correction (on by default). It can be turned off by specifying
-``apply_wavecorr=False`` when running the step.
+The WAVECORR reference file contains pixel offset values as a function of
+wavelength and source offset within a NIRSpec slit.
+It is used when applying the NIRSpec wavelength zero-point correction to
+fixed-slit (EXP_TYPE="NRS_FIXEDSLIT"), bright object TSO
+(EXP_TYPE="NRS_BRIGHTOBJ"), and MSA/MOS spectra (EXP_TYPE="NRS_MSASPEC").
+This is an optional correction that is turned on by default.
+It can be turned off by specifying ``apply_wavecorr=False`` when running the step.
 
 .. include:: wavecorr_selection.rst
 
-WAVELENGTHRANGE Reference
--------------------------
+.. include:: ../includes/standard_keywords.rst
+
+Type Specific Keywords for WAVECORR
++++++++++++++++++++++++++++++++++++++
```

Comment:
This line looks longer than needed. Might mess up formatting.

Reply:
I don't think it hurts to have them longer than the text, but I can always fix to make sure.